### ISSUE TITLE
SQLSTATE 23505

### ISSUE THREAD

**Issue by iwconfig** [divyam234/teldrive#229]
Howdy!

I'm using `rclone` with VFS mode "full" to mount my drive and when modifying an existing remote file i get the following repeated error message:

```console
2024/05/24 18:28:30 ERROR : file.txt: Failed to copy: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
2024/05/24 18:28:30 ERROR : file.txt: vfs cache: failed to upload try #1, will retry in 10s: vfs cache: failed to transfer file from cache to remote: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
```

and I find this in the teldrive log:

```console
teldrive  | 24/05/2024 04:26 PM ERROR   [DB] github.com/divyam234/teldrive/pkg/services/file.go:101 ERROR: duplicate key value violates unique constraint "unique_file" (SQLSTATE 23505)
```

<details>
  <summary>more teldrive context (spoiler)</summary>

  ```console
  teldrive  | 24/05/2024 04:26 PM DEBUG   uploading file  {"fileName": "file.txt", "partName": "87d1b1ee64da1909af6f798103789725", "bot": "6493112858", "botNo": 0, "chunkNo": 1, "partSize": 2391}
  teldrive  | 24/05/2024 04:26 PM DEBUG   upload finished {"fileName": "file.txt", "partName": "87d1b1ee64da1909af6f798103789725", "chunkNo": 1}
  teldrive  | 24/05/2024 04:26 PM INFO            {"status": 201, "method": "POST", "path": "/api/uploads/ae63c7d83b38586880f8f71368c76e78", "query": "channelId=0&encrypted=false&fileName=file.txt&partName=87d1b1ee64da1909af6f798103789725&partNo=1", "ip": "172.18.0.1", "user-agent": "rclone/v1.66.2", "latency": 1.022449512, "time": "2024-05-24T16:26:49Z"}
  teldrive  | 24/05/2024 04:26 PM ERROR   [DB] github.com/divyam234/teldrive/pkg/services/file.go:101 ERROR: duplicate key value violates unique constraint "unique_file" (SQLSTATE 23505)
  teldrive  | [131.500ms] [rows:0] INSERT INTO "teldrive"."files" ("name","type","mime_type","path","size","starred","depth","category","encrypted","user_id","status","parent_id","parts","channel_id") VALUES ('file.txt','file','application/octet-stream','',2391,false,NULL,'other',false,6907387205,'active','nCkUeR9W6YwAGwVH','[{"id":16254}]',2052950520) RETURNING "id","created_at","updated_at"
  teldrive  | 24/05/2024 04:26 PM ERROR   &{key conflict 409}
  teldrive  | 24/05/2024 04:26 PM ERROR   key conflict
  teldrive  | 24/05/2024 04:26 PM INFO            {"status": 409, "method": "POST", "path": "/api/files", "query": "", "ip": "172.18.0.1", "user-agent": "rclone/v1.66.2", "latency": 0.547395099, "time": "2024-05-24T16:26:49Z"}
  teldrive  | 24/05/2024 04:26 PM DEBUG   uploading file  {"fileName": "file.txt", "partName": "7bfd43fb463367ddc71a0ecbb653da27", "bot": "7192771920", "botNo": 1, "chunkNo": 1, "partSize": 2391}
  teldrive  | 24/05/2024 04:26 PM DEBUG   upload finished {"fileName": "file.txt", "partName": "7bfd43fb463367ddc71a0ecbb653da27", "chunkNo": 1}
  teldrive  | 24/05/2024 04:26 PM INFO            {"status": 201, "method": "POST", "path": "/api/uploads/ae63c7d83b38586880f8f71368c76e78", "query": "channelId=0&encrypted=false&fileName=file.txt&partName=7bfd43fb463367ddc71a0ecbb653da27&partNo=1", "ip": "172.18.0.1", "user-agent": "rclone/v1.66.2", "latency": 0.88459958, "time": "2024-05-24T16:26:51Z"}
  teldrive  | 24/05/2024 04:26 PM ERROR   [DB] github.com/divyam234/teldrive/pkg/services/file.go:101 ERROR: duplicate key value violates unique constraint "unique_file" (SQLSTATE 23505)
  teldrive  | [129.213ms] [rows:0] INSERT INTO "teldrive"."files" ("name","type","mime_type","path","size","starred","depth","category","encrypted","user_id","status","parent_id","parts","channel_id") VALUES ('file.txt','file','application/octet-stream','',2391,false,NULL,'other',false,6907387205,'active','nCkUeR9W6YwAGwVH','[{"id":16255}]',2052950520) RETURNING "id","created_at","updated_at"
  teldrive  | 24/05/2024 04:26 PM ERROR   &{key conflict 409}
  teldrive  | 24/05/2024 04:26 PM ERROR   key conflict
  ```

</details>

I have no idea what this really means. I guess it is the same as trying to upload the same file again via the UI. It gives "file exist" error.

<details>
  <summary>Same with `rclone sync/copy --delete-before` (spoiler)</summary>

  ```console
  ~$ teldrive-rclone sync --delete-before file.txt teldrive:
  2024/05/24 19:38:05 ERROR : file.txt: Failed to copy: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  2024/05/24 19:38:05 ERROR : Attempt 1/3 failed with 1 errors and: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  2024/05/24 19:38:07 ERROR : file.txt: Failed to copy: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  2024/05/24 19:38:07 ERROR : Attempt 2/3 failed with 1 errors and: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  2024/05/24 19:38:09 ERROR : file.txt: Failed to copy: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  2024/05/24 19:38:09 ERROR : Attempt 3/3 failed with 1 errors and: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  2024/05/24 19:38:09 Failed to sync: HTTP error 409 (409 Conflict) returned body: "{\"code\":409,\"message\":\"key conflict\"}"
  ```  

</details>

Is it not possible to modify files? Is it not possible to delete the remote file before uploading the new, modified, file automatically? That's what I do manually, which is a rather tedious task.

Or is it a problem with my database?

**Comment by dogemanttv** [comment]
try deleting the rclone folder in ~/.cache

**Comment by iwconfig** [comment]
> try deleting the rclone folder in ~/.cache

No rclone folder in there, but by "`~/.cache`" you actually meant to say deleting the VFS cache folder then no, it does not solve the issue which I know because yes, I have done that already. I use `--cache-dir`. Thanks for the suggestion though! 🙂

I don't think `rclone` is the issue really, at least not on my part, or it might be the teldrive implementation in it.

**Comment by iwconfig** [comment]
**Oh wonderful!** The issue is fixed in the `teldrive` v1.3.0 and `divyam234/rclone` v1.66.3 release!

# Thank you so much @divyam234! 💯 😄
